tests/ui/type-alias-impl-trait/closure_wf_outlives.rs

```diff
@@ -12,27 +12,27 @@
 // requires `'a: 'b` bound
 mod test1 {
     type Opaque<'a, 'b> = impl Sized + 'a + 'b;
-    //~^ ERROR lifetime bound not satisfied
 
     fn define<'a, 'b>() -> Opaque<'a, 'b>
     where
         'a: 'b,
     {
         || {}
+        //~^ ERROR lifetime bound not satisfied
```

Comment:
This is as bad as outlives errors we get within a defining scope, like:

```rust
fn foo<'a, T>(t: T) -> impl Sized + 'a { t }
```

```
error[E0309]: the parameter type `T` may not live long enough
 --> src/lib.rs:1:42
  |
1 | fn foo<'a, T>(t: T) -> impl Sized + 'a { t }
  |        --                                ^ ...so that the type `T` will meet its required lifetime bounds
  |        |
  |        the parameter type `T` must be valid for the lifetime `'a` as defined here...
```

Reply:
can we also emit the "this definition site has more where clauses than the opaque type" note for region errors, I feel like it's even more useful there

Reply:
really annoying to do that given the way that region error reporting is handled with a totally different error path :/